substrate/primitives/staking/src/lib.rs

```diff
@@ -104,6 +104,21 @@ impl<Balance: Copy> From<Balance> for Stake<Balance> {
 	}
 }
 
+/// Represents the reason for a stake update.
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum StakeUpdateReason {
+	/// Stake update due to a bond, bonding extra or unbonding.
+	Bond,
+	/// Stake update due to a validator staking reward.
+	ValidatorReward,
```

Comment:
Ankan: `ValidatorPayout`, etc

And in the future we can have e.g. `NominatorSinglePayout`